### PR TITLE
Abhinandsuresh

### PR DESCRIPTION
**Pull request by AbhinandSuresh**
Water level monitoring module code updated

### CODE REVIEW

File: AWMSmonitor.ino [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+   Author: Mr.Greatin Baby and Mr. Abhinand Suresh
+   Code for water level monitoring module
+   LowPower library is used for putting arduino into sleep mode
+   Communication is done using radio wave transmission of packets.
+   LoRa modules are used for transmitting datas.
+*/
+#include <LowPower.h>
+#include <SPI.h>
+#include <LoRa.h>
+#include<String.h>
```
Comment (greatinbaby) [inline AWMSmonitor.ino:14]: space

File: AWMSmonitor.ino [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        if(DEBUG_LEVEL <= level) \
+        Serial.print(msg); \
+        }
+#else
+#define debug_Println(x)
+#define debug_Print(x)
+#endif
+
+int sec = 1;                    //default sleep delay seconds
+byte localAddress = 0xFF;       // address of this device
```
Comment (greatinbaby) [inline AWMSmonitor.ino:51]: look like this is broadcast

File: AWMSmonitor.ino [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+#include <LowPower.h>
+#include <SPI.h>
+#include <LoRa.h>
+#include<String.h>
+#define ECHO_PIN 5            // Echo pin of Ultrasonic sensor
+#define TRIGGER_PIN 6         // Trigger pin of Ultrasonic sensor
+#define LORA_FREQUENCY 433E6  // Set LoRa frequency 
+#define BAUD_RATE 9600        // Serial baud rate
+#define onAck "mon"           // Expected Acknowledgement for motor power on (May be updated after cyphering)
+#define offAck "mof"          // Expected Acknowledgement for motor power off (May be updated after cyphering)
```
Comment (greatinbaby) [inline AWMSmonitor.ino:20]: TODO: Please form a messaging style reply
may be A facy number ( 2bytes): indented client #, 1 byte, status(1 byte), message, message parameters etc. 
For example : AABB, 1A, 01 ( WORKING/ 02, sleeping etc), 05(MOTOR status - 00-normal ACK, 01, configuration, 02 - PAIRING, 03 - TIME, 04, TIME request etc), 01(motor ON)

File: AWMSmonitor.ino [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        Serial.print(msg); \
+        }
+#else
+#define debug_Println(x)
+#define debug_Print(x)
+#endif
+
+int sec = 1;                    //default sleep delay seconds
+byte localAddress = 0xFF;       // address of this device
+byte destination = 0x80;        // destination to send to
```
Comment (greatinbaby) [inline AWMSmonitor.ino:52]: should be assigned while pairing to client(s)

File: AWMSmonitor.ino [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+String acknowledgment;
+long duration;
+float level;
+byte count = 0;
+String device_id = "IoTuneZ_906: "; // a fancy device id
```
Comment (greatinbaby) [inline AWMSmonitor.ino:57]: dont use any strings here

File: AWMSmonitor.ino [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+  if (!LoRa.begin(LORA_FREQUENCY))
+    {
+      debug_Println(0,"Starting LoRa failed!");
+      while (1);
+    }
+}
+/*Function for arduino board to go to sleep for some seconds*/
+void system_sleep(int sleepVal)
+{
+  delay(5);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:74]: why this delay?

File: AWMSmonitor.ino [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+  digitalWrite(TRIGGER_PIN, HIGH);
+  delayMicroseconds(8);
+  digitalWrite(TRIGGER_PIN, LOW);
+  // Reads the echo Pin, returns the sound wave travel time in microseconds
+  duration = pulseIn(ECHO_PIN, HIGH);
+  // Calculating the level
+  level = duration * 0.034 / 2; // Speed of sound wave divided by 2 (go and back)
+  //delay(100);
+  String LoRaMessage = String(device_id) + String(count) + ">" + String(level);
+  //delay(50);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:106]: remove unwanted comments

File: AWMSmonitor.ino [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+{
+  LoRa_txMode();                        // set tx mode
+  LoRa.beginPacket();                   // start packet
+  LoRa.print(message);                  // add payload
+  LoRa.endPacket(true);                 // finish packet and send it
+}
+
+void sendData()
+{
+  digitalWrite(TRIGGER_PIN, LOW);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:94]: calculating distance and sending are two different functionalities
what is data means?

File: AWMSmonitor.ino [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  //delay(50);
+  debug_Println(2,"Sending " + LoRaMessage + " to 0x" + (destination));
+  LoRa_sendMessage(LoRaMessage);
+  count++;  //increment message ID
+}
+void loraAck_wait()
+{
+  //Receive Ack
+  //wait for Acknoledgement
+  int count = 5000; //to wait for 5 seconds (5000 ms)
```
Comment (greatinbaby) [inline AWMSmonitor.ino:115]: #define wait period

File: AWMSmonitor.ino [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+void on_receive(int packetSize)
+{
+  String incoming = "";
+  while (LoRa.available())
+  {
+    incoming += (char)LoRa.read();
+  }
+  //delay(400);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:145]: pls remove

File: AWMSmonitor.ino [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+  // if message is for this device, or broadcast, print details:
+  debug_Print(2,"Message: ");
+  debug_Println(2,incoming);
+  debug_Println(2,"RSSI: " + String(LoRa.packetRssi()));
+  debug_Println(2,"Snr: " + String(LoRa.packetSnr()));
+  isAcknoledged = true;
+  //isPktAvailable = true;
```
Comment (greatinbaby) [inline AWMSmonitor.ino:152]: remove

File: AWMSmonitor.ino [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+    sec = NO_ACK_SLEEP;
+  }
+  LoRa_txMode();
+}
+
+void LoRa_rxMode()
+{
+  LoRa.receive();
+  debug_Println(1,"rxMode");
+  //delay(5);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:172]: :@

File: AWMSmonitor.ino [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+void LoRa_txMode()
+{
+  LoRa.idle();                          // set standby mode
+  LoRa.disableInvertIQ();               //tx mode
+  debug_Println(1,"txMode");
+  delay(5);
```
Comment (greatinbaby) [inline AWMSmonitor.ino:180]: why this delay

**Review by greatinbaby — CHANGES_REQUESTED** [review]

File: AWMSmonitor.ino (hunk 7 above)
Comment (AbhinandSuresh) [reply AWMSmonitor.ino:106]: ok

File: AWMSmonitor.ino (hunk 1 above)
Comment (AbhinandSuresh) [reply AWMSmonitor.ino:14]: ok

File: AWMSmonitor.ino (hunk 5 above)
Comment (AbhinandSuresh) [reply AWMSmonitor.ino:57]: ok

File: AWMSmonitor.ino (hunk 8 above)
Comment (AbhinandSuresh) [reply AWMSmonitor.ino:94]: ok, Data here is the level monitor data (read from the ultrasonic sensor) in a formatted way (including device ID, Message, 
 and Message ID)

File: AWMSmonitor.ino (hunk 9 above)
Comment (AbhinandSuresh) [reply AWMSmonitor.ino:115]: ok

**Pull request closed without merging**